### PR TITLE
AWSI automation tests support on Linux

### PR DESCRIPTION
**Pull request by junbo75**
Signed-off-by: Junbo Liang <68558268+junbo75@users.noreply.github.com>

- Ran the scripts on Linux to deploy and destroy CDK applications
- Regenerated all the test levels to use prefab
- Automation tests passed on Linux

Review should focus on the following files since other changes are recreating the levels and move files around for sharing between Windows and Linux:
- AutomatedTesting/Gem/PythonTests/AWS/README.md 
- scripts/build/Platform/Linux/deploy_cdk_applications.sh 
- scripts/build/Platform/Linux/deploy_cdk_applications.sh

### CODE REVIEW

File: AutomatedTesting/Gem/PythonTests/AWS/README.md [hunk 1]
```diff
@@ -2,30 +2,61 @@
...
+    * Windows
+        ```
+            {engine_root}\scripts\build\Platform\Windows\deploy_cdk_applications.cmd
+        ```
+    * Linux
+        ```
+            {engine_root}/scripts/build/Platform/Linux/deploy_cdk_applications.sh
+        ```
+3. Open a new CLI window at the engine root and set the following environment variables:
+    * Windows
```
Comment (allisaurus) [inline AutomatedTesting/Gem/PythonTests/AWS/README.md:21]: suggest specifying "Windows **(command prompt)**", as the syntax for setting env vars in PowerShell (many Windows users' first choice of terminal) is different.

File: scripts/build/Platform/Linux/destroy_cdk_applications.sh [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+{
+# Destroy the CDK application for a specific AWS gem
+GEM_NAME=$1
+echo [cdk_destruction] Destroy the CDK application for the $GEM_NAME gem
+pushd $GEM_DIRECTORY/$GEM_NAME/cdk
+
+# Revert the CDK application code to a stable state using the provided commit ID
+if ! git checkout $COMMIT_ID -- .;
+then
+    echo [git_checkout] Failed to checkout the CDK application for the %GEM_NAME% gem using commit ID $COMMIT_ID
```
Comment (allisaurus) [inline scripts/build/Platform/Linux/destroy_cdk_applications.sh:28]: should be `$GEM_NAME` (not Windows-y `%GEM_NAME%`)

File: scripts/build/Platform/Linux/deploy_cdk_applications.sh [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    # Deploy the CDK application for a specific AWS gem
+    GEM_NAME=$1
+    ADDITIONAL_ARGUMENTS=$2
+    echo [cdk_deployment] Deploy the CDK application for the $GEM_NAME gem
+    pushd $GEM_DIRECTORY/$GEM_NAME/cdk
+
+    # Revert the CDK application code to a stable state using the provided commit ID
+    if ! git checkout $COMMIT_ID -- .;
+    then
+        echo [git_checkout] Failed to checkout the CDK application for the %GEM_NAME% gem using commit ID $COMMIT_ID
```
Comment (allisaurus) [inline scripts/build/Platform/Linux/deploy_cdk_applications.sh:29]: same comment as for destroy_cdk_applications.sh L28 (use `$GEM_NAME`)

**Review by onecent1101 — COMMENTED** [review]
Assume it is just a "converted" version from windows script, just address comments, it is good to go.

File: AutomatedTesting/Gem/PythonTests/AWS/README.md (hunk 1 above)
Comment (junbo75) [reply AutomatedTesting/Gem/PythonTests/AWS/README.md:21]: Sure. I'll add that.

File: scripts/build/Platform/Linux/destroy_cdk_applications.sh (hunk 2 above)
Comment (junbo75) [reply scripts/build/Platform/Linux/destroy_cdk_applications.sh:28]: Nice catch! Will make the change.

**Review by allisaurus — APPROVED** [review]

File: AutomatedTesting/Gem/PythonTests/AWS/aws_metrics/aws_metrics_automation_test.py [hunk 4]
```diff
@@ -1,289 +1,289 @@
...
+@pytest.mark.parametrize('project', ['AutomatedTesting'])
+@pytest.mark.parametrize('region_name', [constants.AWS_REGION])
+@pytest.mark.parametrize('resource_mappings_filename', [constants.AWS_RESOURCE_MAPPING_FILE_NAME])
+@pytest.mark.parametrize('session_name', [constants.SESSION_NAME])
+@pytest.mark.parametrize('stacks', [[f'{constants.AWS_PROJECT_NAME}-{AWS_METRICS_FEATURE_NAME}-{constants.AWS_REGION}']])
+class TestAWSMetricsWindows(object):
+    """
+    Test class to verify the real-time and batch analytics for metrics.
+    """
+    @pytest.mark.parametrize('level', ['levels/aws/metrics/metrics.spawnable'])
```
Comment (lmbr-pip) [inline AutomatedTesting/Gem/PythonTests/AWS/aws_metrics/aws_metrics_automation_test.py:153]: Can you speak to this name change? I don't see a metrics.spawnable like file in this PR

Comment (junbo75) [reply AutomatedTesting/Gem/PythonTests/AWS/aws_metrics/aws_metrics_automation_test.py:153]: The .spawnable file is the product assets of the prefab file and locates in the cache. The level is not loaded from source assets.

**Review by shirangj — APPROVED** [review]

**Pull request merged**